src/main/java/graphql/schema/idl/SchemaGenerator.java

```diff
@@ -694,17 +697,32 @@ private GraphQLScalarType buildScalar(BuildContext buildCtx, ScalarTypeDefinitio
             scalar = scalar.transform(builder -> builder
                     .definition(typeDefinition)
                     .comparatorRegistry(buildCtx.getComparatorRegistry())
-                    .withDirectives(
-                            buildDirectives(typeDefinition.getDirectives(),
-                                    directivesOf(extensions), SCALAR, buildCtx.getDirectiveDefinitions(), buildCtx.getComparatorRegistry())
+                    .specifiedByUrl(getSpecifiedByUrl(typeDefinition, extensions))
+                    .withDirectives(buildDirectives(
+                            typeDefinition.getDirectives(),
+                            directivesOf(extensions),
+                            SCALAR,
+                            buildCtx.getDirectiveDefinitions(),
+                            buildCtx.getComparatorRegistry())
                     ));
-            //
-            // only allow modification of custom scalars
             scalar = directiveBehaviour.onScalar(scalar, buildCtx.mkBehaviourParams());
         }
         return scalar;
     }
 
+    private String getSpecifiedByUrl(ScalarTypeDefinition scalarTypeDefinition, List<ScalarTypeExtensionDefinition> extensions) {
+        List<Directive> allDirectives = new ArrayList<>(scalarTypeDefinition.getDirectives());
+        extensions.forEach(extension -> allDirectives.addAll(extension.getDirectives()));
+        Optional<Directive> specifiedByDirective = FpKit.findOne(allDirectives,
+                directiveDefinition -> directiveDefinition.getName().equals(Directives.SpecifiedByDirective.getName()));
+        if (!specifiedByDirective.isPresent()) {
+            return null;
+        }
+        Argument urlArgument = specifiedByDirective.get().getArgument("url");
+        StringValue url = (StringValue) urlArgument.getValue();
+        return url.getValue();
```

Comment:
Thought - Will it always be non null arg values?

Reply:
yes it is: url is required non null